### DISCUSSION TITLE
Speed up "cold start"

### DISCUSSION THREAD

**Discussion by simplenotezy** [jmcdo29/nest-commander#1069] [Ideas]
Can anything be done to reduce the latency from when you hit enter to when the command is run? Not sure if SWC or Vite could help at all, but they have fast "cold" starts.

Example:

https://github.com/jmcdo29/nest-commander/assets/3786627/47eee28a-8af6-4857-8fe4-6e655cf5f632

**Comment by jmcdo29** [comment]
The reason for the latency is that a nest application has to be created. Nest can be a bit slow creating three DI tree and application because of all the steps it needs to take. Any database connection you say is necessary has to be connected, which can also add to overhead. You can use lazy module loading to speed things up a bit, but there's nothing nest-commander can easily do about it
